rust/lance/src/dataset.rs

```diff
@@ -1605,6 +1651,30 @@ impl Dataset {
         Ok(())
     }
 
+    pub async fn shallow_clone(
+        &mut self,
+        target_path: &str,
+        ref_name: &str,
+        store_params: ObjectStoreParams,
+    ) -> Result<Self> {
+        // self.tags.create(ref_name, self.version().version).await?;
```

Comment:
remove this line? Or I think we can have an optional flag of `create_tag` which can run this if equal to true.